tests/mu_substitute_test.c

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "test_helper.h"
+
+#include "azure_macro_utils/macro_utils.h"
+
+#include "mu_substitute_test.h"
+
+#define DIF(x,y) ((x)-(y))
+#define MUL(x,y) ((x)*(y))
+
+#define MU_IS_SUBST_PAIR
+
+
+#define DO_MUL_IN_ARRAY(X) MU_SUBSTITUTE_L3(X, MUL) , /*note: a comma is needed between the array elements */
```

Comment:
>MU_SUBSTITUTE_L3 [](start = 27, length = 16)

So this can't be L1 because there is already L1 on line 21? Or because all of the MU_SUBSTITUTE_L macros try to substitute PAIR?

Reply:
In this instance I could have used L1 everywhere in this file and it would all be fine.

When multiple of the L1 would be chained, then it becomes a problem and different names are needed.

---
In reply to: [508002637](https://github.com/Azure/azure-macro-utils-c/pull/49#discussion_r508002637) [](ancestors = 508002637)